### PR TITLE
Build: Add package-lock.json back, remove it to .gitignore

### PR DESCRIPTION
**Pull request by posebear1990**
npm 5.6.0 has fixed optionalDependencies bug

Ref #3792

### Summary ###

#3792 
remove package-lock.json 
because npm bug has been fixed in 5.6.0， see：
https://github.com/npm/npm/issues/18135

### Checklist ###

* [x] All authors have signed the CLA at https://cla.js.foundation/jquery/jquery
* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

**Comment by mgol** [comment]
Thanks for the PR. Have you tested that `package-lock.json` is no longer modified on Linux when doing `npm install`? The issue npm issue https://github.com/npm/npm/issues/17722 is still open and [this comment](https://github.com/npm/npm/issues/17722#issuecomment-352487565) suggests it's not fixed.

File: .gitignore [hunk 1]
```diff
@@ -8,7 +8,6 @@
 .bower.json
 .sizecache.json
 yarn.lock
-package-lock.json
```
Comment (mgol) [inline .gitignore]: You haven't actually added the file to the repo, you've just removed it from `.gitignore`. This is only a part of what's needed here.

Comment (posebear1990) [reply .gitignore]: I have made test in my ubuntu 17.04, I'm not sure if the other distribution will has this issue again.

**Comment by mgol** [comment]
Thanks for the PR but we will wait a little for the situation around npm lockfile to stabilize; people still report issues on https://github.com/npm/npm/issues/17722 which is still open.

**Pull request closed without merging**